src/main/java/mate/academy/spring/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.spring.security;
+
+import java.util.Optional;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> userOptional = userService.findByEmail(email);
+        if (userOptional.isEmpty()) {
+            throw new UsernameNotFoundException("User not found");
+        }
+        User user = userOptional.get();
+        UserBuilder userBuilder =
+                org.springframework.security.core.userdetails.User.withUsername(email);
+        userBuilder.password(user.getPassword());
+        userBuilder.authorities(user.getRoles().stream()
```

Comment:
```suggestion
        userBuilder.roles(user.getRoles().stream()
```
we talked about the difference today